src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,48 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            MachineProducer<Bulldozer> producer = new BulldozerProducer();
+            List<Bulldozer> bulldozerList = producer.get();
+            return new ArrayList<>(bulldozerList);
+        }
+        if (type == Excavator.class) {
+            MachineProducer<Excavator> producer = new ExcavatorProducer();
+            List<Excavator> excavatorList = producer.get();
+            return new ArrayList<>(excavatorList);
+        }
+        if (type == Truck.class) {
+            MachineProducer<Truck> producer = new TruckProducer();
+            List<Truck> truckList = producer.get();
+            return new ArrayList<>(truckList);
```

Comment:
producers should be class-level `private final` fields
```suggestion
            return bulldozerProducer.get();
        }
        if (type == Excavator.class) {
            return excavatorProducer.get();
        }
        if (type == Truck.class) {
            return truckProducer.get();
```